src/include/duckdb/logging/logging.hpp

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/logging/log_storage.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/atomic.hpp"
+#include "duckdb/common/mutex.hpp"
+#include "duckdb/common/optional_idx.hpp"
+#include "duckdb/common/types.hpp"
+#include "duckdb/common/unordered_set.hpp"
+#include "duckdb/common/typedefs.hpp"
+
+namespace duckdb {
+
+enum class LogLevel : uint8_t {
```

Comment:
I understand the idea behind log levels, but its not clear what a `FATAL` would even be in DuckDB world, or what the difference between `TRACE` and `DEBUG` should be